dotCMS/src/main/java/com/dotmarketing/filters/CMSFilter.java

```diff
@@ -203,6 +200,23 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
         chain.doFilter(req, res);
     }
 
+    private IAm resolveResourceType(final IAm iAm,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)